lib/versionmanager.js

```diff
@@ -241,12 +242,28 @@ function upgradePackageDefinitions(currentDependencies, options) {
  * @param {Object} [options={}]
  * @returns {string} The updated package data, as utf8 text
  */
-function upgradePackageData(pkgData, oldDependencies, newDependencies, newVersions, options = {}) {
+async function upgradePackageData(pkgData, oldDependencies, newDependencies, newVersions, options = {}) {
     for (const dependency in newDependencies) {
         if (!options.minimal || !isSatisfied(newVersions[dependency], oldDependencies[dependency])) {
+            if (options.upgrade && options.interactive) {
+                const to = versionUtil.colorizeDiff(newDependencies[dependency] || '', oldDependencies[dependency]);
+                const q = {
+                    type: 'confirm',
+                    name: 'value',
+                    message: `Do you want to upgrade: ${dependency} ${oldDependencies[dependency]} → ${to}?`,
+                    initial: true
+                };
+                let response = await prompts(q);
+                if (!response.value) {
+                    delete newDependencies[dependency];
+                    delete oldDependencies[dependency];
+                    continue;
+                }
+            }
             const expression = `"${dependency}"\\s*:\\s*"${escapeRegexp(`${oldDependencies[dependency]}"`)}`;
             const regExp = new RegExp(expression, 'g');
             pkgData = pkgData.replace(regExp, `"${dependency}": "${newDependencies[dependency]}"`);
```

Comment:
While we are at it, we could return `pkgData` instead of mutating it as well.

Reply:
I Don't quite understand what you mean.

Reply:
Nvm, I see now it's just overwriting the reference, not mutating the argument.